### ISSUE TITLE
ZVV: de.schildbach.pte.exception.NotFoundException: https://online.fahrplan.zvv.ch/bin/mgate.exe

### ISSUE THREAD

**Issue by Baerbeisser** [schildbach/public-transport-enabler#606]
### Query
- NetworkId: `ZVV`
- From: `Location{STATION, 8505000, 47.0501790/8.3101860, name=Luzern, products=[REGIONAL_TRAIN, SUBURBAN_TRAIN, BUS, FERRY]}`
- Via: `null`
- To: `Location{STATION, 8506000, 47.5004210/8.7233750, name=Winterthur, products=[REGIONAL_TRAIN, SUBURBAN_TRAIN, BUS]}`
- Date: `Wed Nov 27 11:42:37 GMT+01:00 2024`
- Departure: `true`
- Products: `[HIGH_SPEED_TRAIN, REGIONAL_TRAIN, SUBURBAN_TRAIN, SUBWAY, TRAM, BUS, FERRY, CABLECAR, ON_DEMAND]`
- Optimize for: `LEAST_CHANGES`
- Walk Speed: `NORMAL`

### Error
```
de.schildbach.pte.exception.NotFoundException: https://online.fahrplan.zvv.ch/bin/mgate.exe
de.schildbach.pte.util.HttpClient.getInputStream(HttpClient.java:296)
de.schildbach.pte.util.HttpClient.get(HttpClient.java:215)
de.schildbach.pte.AbstractHafasClientInterfaceProvider.jsonTripSearch(AbstractHafasClientInterfaceProvider.java:588)
```

### Additional information
mgate.exe? On Android?

**Comment by schildbach** [comment]
The ZVV server is of course *not* running on Android.

**Comment by Baerbeisser** [comment]
Wait, they use Windows Server exposed to the public? 😰

**Comment by schildbach** [comment]
It's not that uncommon, I guess.

**Comment by schildbach** [comment]
Regarding this issue, the ZVV seems to have switched their Hafas to using a proxy at https://app-prod-routing.viadi-zero.ch/v1/hafasproxy/. At the moment, PTE has no support for that API. Maybe the mobile app is still using the raw API and could be reversed?

**Comment by derhuerst** [comment]
`hafas-client` equivalent issue: https://github.com/public-transport/hafas-client/issues/322#issue-2462093566

**Comment by emojized** [comment]
can we create a new provider for https://transport.opendata.ch/ I teached it in PHP but I am not proficient in Java.

**Comment by emojized** [comment]
is this something you planning our should i try it myself? the problem is i was really reliant on öffi here in switzerland, found out even lucerne does not work.

**Comment by lsr1** [comment]
> i was really reliant on öffi here in switzerland
Just came here to say "same here, for Zurich".
However I totally understand that this is a FOSS volunteer project. If I can help with a EUR donation via SEPA, plz contact me.

**Comment by derhuerst** [comment]
[`transport-apis` has been adapted to the latest ZVV endpoint](https://github.com/public-transport/transport-apis/pull/99), so public-transport-enabler could pull the data from there.

cc @schildbach

**Comment by schildbach** [comment]
Thanks! Fixed via 81b1122204481bc980b091961244ecdb1a86aec0.
